monitors/CRMonitor.js

```diff
@@ -0,0 +1,148 @@
+    // If it is, then send an alert of formatted markdown text.
+    for (let wallet of this.walletsToMonitor) {
+      const [shouldPush, crRatio] = this._shouldPushWalletNotification(wallet, priceFeed);
+      if (shouldPush) {
+        // Sample message:
+        // Risk alert: [Tracked wallet name] has fallen below [threshold]%.
+        // Current [name of identifier] value: [current identifier value].
+        const mrkdwn =
+          wallet.name +
+          " (" +
+          createEtherscanLinkMarkdown(this.web3, wallet.address) +
+          ") collateralization ratio has dropped to " +
+          this.formatDecimalString(crRatio) +
+          "% which is below the " +
+          wallet.crAlert +
+          "% threshold. Current value of " +
+          this.syntheticCurrencySymbol +
+          " is " +
+          this.formatDecimalString(priceFeed);
+
+        this.logger.info({
+          at: "ContractMonitor",
+          message: "Collateralization ratio alert 🚨!",
+          mrkdwn: mrkdwn
+        });
+      }
+    }
+  };
+
+  _getPositionInformation = address => {
+    const positionInfo = this.empClient.getAllPositions().filter(position => position.sponsor == address);
+    if (positionInfo.length == 0) {
+      return null;
+      // there should only ever be one position information object per address
+    } else return positionInfo[0];
+  };
+
+  _shouldPushWalletNotification(wallet, priceFeed) {
+    const positionInformation = this._getPositionInformation(wallet.address);
+    if (positionInformation == null) {
+      // There is no position information for the given wallet.
+      return [false, 0];
+    }
+
+    const collateral = positionInformation.amountCollateral;
+    const tokensOutstanding = positionInformation.numTokens;
+
+    // If the values for collateral or price have yet to resolve, dont push a notification
+    if (collateral == null || tokensOutstanding == null) {
+      return [false, 0];
+    }
+
+    // If CR = null then there are no tokens outstanding and so dont push a notification
+    const positionCR = this._calculatePositionCRPercent(collateral, tokensOutstanding, priceFeed);
+    if (positionCR == null) {
+      return [false, 0];
+    }
+
+    let shouldPushWalletNotification = false;
+    if (this._ltThreshold(positionCR, this.web3.utils.toWei(wallet.crAlert.toString()))) {
+      if (!this.walletsAlerted[wallet.address].crAlert) {
+        shouldPushWalletNotification = true;
+      }
+      this.walletsAlerted[wallet.address].crAlert = true;
+    } else {
+      this.walletsAlerted[wallet.address].crAlert = false;
+    }
+    return [shouldPushWalletNotification, positionCR];
+  }
+
+  // Checks if a big number value is below a given threshold.
+  _ltThreshold(value, threshold) {
+    // If the price has not resolved yet then return false.
+    if (value == null) {
+      return false;
+    }
+    return this.web3.utils.toBN(value).lt(this.web3.utils.toBN(threshold));
+  }
```

Comment:
This code looks pretty similar to code introduced in #1336. Is there a way we could abstract some of this to another object or library? Added advantage is that we can test it once and not have to re-test it in both modules.